### PR TITLE
Remove intl polyfills

### PR DESCRIPTION
**Pull request by samdark**
| Q             | A
| ------------- | ---
| Is bugfix?    | no
| New feature?  | no
| Breaks BC?    | yes
| Tests pass?   | yes
| Fixed issues  | #11397

### CODE REVIEW

File: framework/UPGRADE.md [hunk 1]
```diff
@@ -177,6 +177,11 @@ Upgrade from Yii 2.0.x
   this package manually for your project.
 * `yii\BaseYii::powered()` method has been removed. Please add "Powered by Yii" link either right into HTML or using
   `yii\helpers\Html::a()`.
+* `yii\i18n\MessageFormatter` no longer supports parameter names with `.`, `-`, `=` and other symbols that are used in
+  pattern syntax following directly how it works in intl/ICU. If you use such parameters names, replace special
+  symbols with `_`.
+* `yii\i18n\MessageFormatter::parse()` method was removed. If you have a rare case where it's used copy-paste it from
```
Comment (SilverFire) [inline framework/UPGRADE.md]: Ehh, doesn't seem like a good advise.

File: framework/behaviors/OptimisticLockBehavior.php [hunk 2]
```diff
@@ -32,7 +32,7 @@
  * public function behaviors()
  * {
  *     return [
- *         OptimisticLockBehavior::className(),
+ *         OptimisticLockBehavior::class,
```
Comment (SilverFire) [inline framework/behaviors/OptimisticLockBehavior.php]: Fixed in https://github.com/yiisoft/yii2/commit/be8633702d50eee2fecb9158aa877afbd63f8b7b, update branch from 2.1

**Review by SilverFire — CHANGES_REQUESTED** [review]
Otherwise good

File: framework/UPGRADE.md (hunk 1 above)
Comment (cebe) [reply framework/UPGRADE.md]: what's wrong with copying it from the framework? If you need it you can maintain it yourself.

**Review by dmirogin — APPROVED** [review]

**Pull request merged**